src/UglyToad.PdfPig.Tokens/InlineImageDataToken.cs

```diff
@@ -18,5 +18,30 @@ public InlineImageDataToken(IReadOnlyList<byte> data)
         {
             Data = data;
         }
+
+        /// <inheritdoc />
+        public bool Equals(IToken obj)
+        {
+            if (this == obj)
+                return true;
+
+            if (!(obj is InlineImageDataToken other))
+            {
+                return false;
+            }
+
+            if (Data.Count != other.Data.Count)
+                return false;
+
+            // Note: This maybe slow?
+            // Maybe pre calculate some sort of hash and compare that?
+            for (var index = 0; index < Data.Count; ++index)
```

Comment:
Like the comment say, should calculate some sort of hash? To avoid full bytes comparison?

Reply:
This is fine I think, we don't actually expose this token anywhere that the `Equals` method will be called with any regularity. Since you'd have to do a full scan for hash collisions anyway I'd leave it as is.